sql/src/main/java/io/crate/metadata/functions/SignatureBinder.java

```diff
@@ -386,18 +386,59 @@ private static void addVarArgTypeSignature(TypeSignature typeSignature,
                                                Map<String, TypeVariableConstraint> typeVariableConstraints,
                                                List<TypeSignature> builder,
                                                int actualArity) {
-        TypeVariableConstraint typeVariableConstraint = typeVariableConstraints.get(typeSignature.getBaseTypeName());
+        TypeVariableConstraint typeVariableConstraint = resolveTypeVariableConstraint(
+            typeSignature,
+            typeVariableConstraints
+        );
         if (typeVariableConstraint != null && typeVariableConstraint.isAnyAllowed()) {
             // Type variables defaults to be bound to the same type.
             // To support independent variable type arguments, each vararg must be bound to a dedicated type variable.
-            String constraintName = "_generated_" + typeSignature.getBaseTypeName() + actualArity;
-            TypeSignature newTypeSignature = new TypeSignature(constraintName);
-            typeVariableConstraints.put(constraintName, typeVariableOfAnyType(constraintName));
+            var newConstraintName = "_generated_" + typeVariableConstraint + actualArity;
+            TypeSignature newTypeSignature;
+            if (typeSignature.getParameters().isEmpty()) {
+                newTypeSignature = new TypeSignature(newConstraintName);
```

Comment:
wouldn't this case result in the same as here? https://github.com/crate/crate/pull/9903/files#diff-f5c59a6293ee86d741b614ade3f78ed4R434
So we may could just always use `replaceTypeVariable()`?